### PR TITLE
DR2-1801 Transform XIP output

### PR DESCRIPTION
**Pull request by MancunianSam**
Preservica return XMl in this form from their API

```xml
<Link linkType="DerivedFrom" linkDirection="To" title="Title.docx" ref="649b2628-3a89-4eb3-9a50-2a6780ee439b" type="CO" apiId="38f17325c262242c9bdbf877db678e93">https://tna.preservica.com/api/entity/content-objects/649b2628-3a89-4eb3-9a50-2a6780ee439b</Link>
```

What the XIP schema says we need is

```xml
<xip:Link>
    <xip:Type>DerivedFrom</xip:Type>
    <xip:FromEntity>UUID OF CURRENT ENTITY</xip:FromEntity>
    <xip:ToEntity>649b2628-3a89-4eb3-9a50-2a6780ee439b</xip:ToEntity>
</xip:Link>
```

We need to transform the first one into the second one.

### CODE REVIEW

**Review by techncl — COMMENTED** [review]
Looks good! Just some minor comments; feel free to ignore.

File: src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala [hunk 1]
```diff
@@ -377,7 +377,33 @@ class DataProcessor[F[_]]()(using me: MonadError[F, Throwable]) {
...
-  def getEntityLinksXml(elem: Elem): F[Seq[Node]] = me.pure((elem \ "Links" \ "Link"))
+  def getEntityLinksXml(ref: UUID, elem: Elem): F[Seq[Node]] = me.pure {
+    val links = elem \ "Links" \ "Link"
+    links.flatMap { link =>
+      for {
+        linkType <- link.attribute("linkType").flatMap(_.headOption)
+        linkDirection <- link.attribute("linkDirection").flatMap(_.headOption)
+        linkRef <- link.attribute("ref").flatMap(_.headOption)
+      } yield {
+        val entities =
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala:388]: `refLinks` is good. I've changed it.

File: src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala [hunk 2]
```diff
@@ -377,7 +377,33 @@ class DataProcessor[F[_]]()(using me: MonadError[F, Throwable]) {
     * @return
     *   A `Seq` of `Node` containing the links
     */
-  def getEntityLinksXml(elem: Elem): F[Seq[Node]] = me.pure((elem \ "Links" \ "Link"))
+  def getEntityLinksXml(ref: UUID, elem: Elem): F[Seq[Node]] = me.pure {
+    val links = elem \ "Links" \ "Link"
+    links.flatMap { link =>
+      for {
+        linkType <- link.attribute("linkType").flatMap(_.headOption)
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala:384]: I'm not calling .toString on the `linkType` Do you mean the `refLinks` ?

[the remark this answers is not in the document]
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala:384]: Sorry! yes, `linkRef`

[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala:384]: I've mapped it toString and removed the toString from below.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**